apps/src/netsim/NetSimTable.js

```diff
@@ -477,18 +487,20 @@ NetSimTable.prototype.updateCacheRow_ = function (id, row) {
 
   if (!_.isEqual(oldRow, newRow)) {
     this.cache_[id] = newRow;
-    this.tableChange.notifyObservers(this.arrayFromCache_());
+    this.tableChange.notifyObservers();
   }
 };
 
 /**
+ * @param {function(key, value)} [predicate] - A condition on returning the row.
```

Comment:
It is not guaranteed to.  [ECMA-262 section 12.6.4 _The for-in statement_](http://www.ecma-international.org/ecma-262/5.1/#sec-12.6.4) states:

> The mechanics and order of enumerating the properties... is not specified.

_For-in_ iteration order is implementation-dependent.  According to [this StackOverflow answer](http://stackoverflow.com/questions/280713/elements-order-in-a-for-in-loop) Chrome and Opera will pull numeric properties in numeric order before any non-numeric properties, but other browsers will pull properties in definition order.

This seems to mean that if we get anything into the cache out-of-order, it will be enumerated in-order on Chrome but out-of-order on Firefox.  This is unlikely, but possible: Suppose one of the partial updates that the log table pulls down is significantly delayed and arrives after another completed request.

We could modify this method to perform a sort and guarantee that it returns a sorted value, but I'd rather just leave that out of its contract and let its clients decide whether they need sorted results.